#!/usr/bin/env python
PACKAGE="dynamic_reconfigure_example"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# The names of the parameters can't be C++ types e.g. double, int, etc
gen.add("double_param", double_t, 1, "frame rate Hz", 20, 0.1, 50)
sub_group = gen.add_group("Sub")  # , type="tab")
sub_group.add("small_double", double_t, 1, "small num test", 0.001, 0.0, 0.5)
sub_group.add("big_double", double_t, 1, "big num test", 5e8, 1000.0, 1.5e9)
gen.add("int_param", int_t, 1, "integer", 50, 1, 256)
gen.add("bool_param", bool_t, 1, "bool", False)
gen.add("str_param", str_t, 1, "string", "test")
gen.add("extra_long_test_name_str", str_t, 1, "string with long label", "test")

size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")

gen.add("enum_param", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_example", "Example"))
